lms/static/js/edxnotes/plugins/llm_summarize.js

```diff
@@ -68,17 +68,113 @@
+                });
+                fetch(request)
+                    .then((response) => {
+                        if (response.ok) return response.json();
+                        throw new Error(gettext('There was an error while summarizing the content.'));
+                    })
+                    .then((data) => {
+                        toggleLoader();
+                        textAreaWrapper.children[0].value = data.summary;
+                    })
+                    .catch((error) => {
+                        alert(error.message);
+                        editor.hide();
+                    });
+
+                function toggleLoader() {
+                    const saveButton = document.querySelector('.annotator-controls .annotator-save');
+                    const loaderWrapper = document.querySelector('.summarize-loader-wrapper');
+                    const textArea = editor.fields[0].element.children[0];
+                    if (textArea.style.display === 'none') textArea.style.display = 'block';
+                    else textArea.style.display = 'none';
+                    loaderWrapper.classList.toggle('show');
+                    saveButton.disabled = !saveButton.disabled;
+                }
+            },
+            cleanupSummarize: function(editor) {
+                const textAreaWrapper = editor.fields[0].element;
+                textAreaWrapper.children[0].value = '';
+                textAreaWrapper.children[1].value = '';
+                editor.element[0].setAttribute('is_summarizing', 'false');
+                const loaderWrapper = document.querySelector('.summarize-loader-wrapper');
+                loaderWrapper.classList.remove('show');
+            },
+            modifyDom: function(annotator) {
                 annotator.adder[0].children[0].id = 'annotateButton';
                 annotator.adder[0].children[0].innerHTML = '<i class="fa fa-pencil" aria-hidden="true"></i>';
                 annotator.adder[0].innerHTML += `
                 <button class="summarize-button" id="summarizeButton" title="${gettext('Summarize text using AI.')}">
                     <i class="fa fa-star" aria-hidden="true"></i>
                 </button>
                 `;
-            },
+                const textAreaWrapper = annotator.editor.fields[0].element;
+                textAreaWrapper.innerHTML += `
+                <div class="summarize-loader-wrapper">
+                    <span class="loader" style="
+                        width: 1.2em;
+                        height: 1.2em;
+                        border: 3px solid rgb(0, 48, 87);
+                        border-bottom-color: white;
+                        border-radius: 50%;
+                        display: inline-block;
+                        box-sizing: border-box;
+                        animation: rotation 3s linear infinite;
+                        margin-right: 5px;
+                        transform-origin: center;">
```

Comment:
Why not add these styles to the stylesheet above. @alexjmpb ?

Reply:
When i apply them on the stylesheet they do not appears. I'm not sure why, but it seems it's out of the scope.